src/builder.rs

```diff
@@ -1296,6 +1299,33 @@ impl<'a, 'gcc, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'gcc, 'tcx> {
 }
 
 impl<'a, 'gcc, 'tcx> Builder<'a, 'gcc, 'tcx> {
+    fn load_inner(&mut self, ptr: RValue<'gcc>, is_volatile: bool) -> RValue<'gcc> {
+        let block = self.llbb();
+        let function = block.get_function();
+
+        // NOTE: instead of returning the dereference here, we have to assign it to a variable in
+        // the current basic block. Otherwise, it could be used in another basic block, causing a
+        // dereference after a drop, for instance.
+        let deref = ptr.dereference(None).to_rvalue();
+        let value_type = deref.get_type();
+
+        // If `is_volatile == true`, convert `ptr` to `volatile value_type *`
+        // and try dereference again. (libgccjit doesn't provide a way to get
```

Comment:
I'm not sure if that's what you're talking about, but an alternative is to do:
```rust
ptr.get_type().get_pointee()
```

Reply:
Thanks, that's exactly what I was looking for!